sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -669,6 +782,27 @@ trait GpuHashJoin extends GpuExec {
         streamedPlan.output, realTarget, joinType, buildSide, compareNullsEqual, spillCallback,
         opTime, joinTime)
     }
+  }
+
+  def doJoin(
+      builtBatch: ColumnarBatch,
+      stream: Iterator[ColumnarBatch],
+      targetSize: Long,
+      spillCallback: SpillCallback,
+      numOutputRows: GpuMetric,
+      joinOutputRows: GpuMetric,
+      numOutputBatches: GpuMetric,
+      opTime: GpuMetric,
+      joinTime: GpuMetric): Iterator[ColumnarBatch] = {
+
+    // The HashJoinIterator takes ownership of the built keys and built data. It will close
+    // them when it is done
+    val joinIterator = if (joinType.isInstanceOf[ExistenceJoin]) {
+      existenceJoinIterator(builtBatch, stream)
```

Comment:
The existence join iterator should update the appropriate metrics, otherwise they will fail to be reported properly.